test/testpartner.ts

```diff
@@ -1,59 +1,119 @@
 
+const DUMP_USER = false
+const DUMP_USER_FILE = './test/fakeUserDump.json'
+
+const asDateStr = (raw: string): Date => new Date(raw)
+
+const asFakeUser = asObject({
+  username: asString,
+  lastLogin: asOptional(asDateStr),
+  loginId: asString,
+  loginKey: asString,
+  repos: asMap(asMap(asUnknown)),
+  server: asUnknown
+})
+
+const asUserDump = asObject({
+  loginKey: asString,
+  data: asFakeUser
+})
+
 async function main(): Promise<void> {
   const allPlugins = {
     bitcoin: makeFakePlugin(btcCurrencyInfo),
     ethereum: makeFakePlugin(ethCurrencyInfo),
+    avalanche: makeFakePlugin(avaxCurrencyInfo),
     ...edgeExchangePlugins
   }
 
   addEdgeCorePlugins(allPlugins)
   lockEdgeCorePlugins()
 
-  const world = await makeFakeEdgeWorld([], {})
+  const fakeUsers: EdgeFakeUser[] = []
+  let loginKey: string = ''
+  if (!DUMP_USER) {
+    const userFile = fs.readFileSync(DUMP_USER_FILE, { encoding: 'utf8' })
+    const json = JSON.parse(userFile)
+    const dump = asUserDump(json)
+    loginKey = dump.loginKey
+    fakeUsers.push(dump.data)
+  }
+
+  const world = await makeFakeEdgeWorld(fakeUsers, {})
   const context = await world.makeEdgeContext({
     apiKey: '',
     appId: '',
     plugins: {
       bitcoin: true,
       ethereum: true,
-      godex: {
-        apiKey: 'XXXXX'
-      }
+      avalanche: true,
+      thorchainda: true
     }
   })
-  const account = await context.createAccount('bob', 'bob123', '1111')
-  const btcWallet = await account.createCurrencyWallet('wallet:bitcoin', {
-    fiatCurrencyCode: 'iso:EUR',
-    name: 'My Fake Bitcoin'
-  })
-  const ethWallet = await account.createCurrencyWallet('wallet:ethereum', {
-    fiatCurrencyCode: 'iso:EUR',
-    name: 'My Fake Bitcoin'
-  })
-  const enabledTokens = ethWallet.currencyInfo.metaTokens.map(
-    token => token.currencyCode
-  )
+  if (DUMP_USER) {
+    const account = await context.createAccount('bob', 'bob123', '1111')
+    await account.createCurrencyWallet('wallet:bitcoin', {
+      fiatCurrencyCode: 'iso:EUR',
+      name: 'My Fake Bitcoin'
+    })
+    const ethWallet = await account.createCurrencyWallet('wallet:ethereum', {
+      fiatCurrencyCode: 'iso:EUR',
+      name: 'My Fake Bitcoin'
+    })
+    const avaxWallet = await account.createCurrencyWallet('wallet:avalanche', {
+      fiatCurrencyCode: 'iso:EUR',
+      name: 'My Fake Avalanche'
+    })
+    const ethEnabledTokens = ethWallet.currencyInfo.metaTokens.map(
+      token => token.currencyCode
+    )
+    await ethWallet.enableTokens(ethEnabledTokens)
+
+    const avaxEnabledTokens = avaxWallet.currencyInfo.metaTokens.map(
+      token => token.currencyCode
+    )
+    await avaxWallet.enableTokens(avaxEnabledTokens)
+
+    const data = await world.dumpFakeUser(account)
+    const dump = {
+      loginKey: account.loginKey,
+      data
+    }
+    fs.writeFileSync(DUMP_USER_FILE, JSON.stringify(dump), { encoding: 'utf8' })
+    process.exit(0)
+  }
```

Comment:
Since setting the boolean completely changes what the program does (building a fake user vs. testing a plugin), should there really be two separate script files, one to test plugins and another to re-build the fake user?

Reply:
done